### PR TITLE
Ophan components name

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

- Parse the DOM in the Ophan component comparator, instead of using regular expressions
- Create a `ophanNestedComponent`, which adds separators automatically
- Keep a running index when grouping cards
- Align some Ophan components name with frontend

## Why?

Working our way towards #4692 

Using a helper–`ophanNestedComponent`–instead of semantic use of whitespace, colons and pipes is an attempt to reduce errors and inconsitencies now and in the future.

Keeping a running index of cards prevents the final index of `news | group-2 | card@3`, `card@3` from not resetting back to zero for each group.

### CODE REVIEW

File: dotcom-rendering/src/components/HeaderTopBarMyAccount.tsx [hunk 1]
```diff
@@ -88,47 +89,57 @@ export const buildIdentityLinks = (
 			id: 'account_overview',
 			url: `${mmaUrl}/`,
 			title: 'Account overview',
-			dataLinkName: 'nav3 : topbar : account overview',
+		},
+		{
+			id: 'billing',
+			url: `${mmaUrl}/billing`,
+			title: 'Billing',
```
Comment (mxdvl) [inline dotcom-rendering/src/components/HeaderTopBarMyAccount.tsx:96]: As per https://github.com/guardian/frontend/blob/95c54cc935f5f4797b53387fc46e421dc43fec1c/common/app/views/support/DropdownMenus.scala#L25-L30

File: dotcom-rendering/src/components/HeaderTopBarMyAccount.tsx [hunk 2]
```diff
@@ -88,47 +89,57 @@ export const buildIdentityLinks = (
...
 		{
 			id: 'email_prefs',
 			url: `${mmaUrl}/email-prefs`,
 			title: 'Emails & marketing',
-			dataLinkName: 'nav3 : topbar : email prefs',
+		},
+		{
+			id: 'data_privacy',
+			url: `${mmaUrl}/data-privacy`,
+			title: 'Data privacy',
```
Comment (mxdvl) [inline dotcom-rendering/src/components/HeaderTopBarMyAccount.tsx:111]: As per https://github.com/guardian/frontend/blob/95c54cc935f5f4797b53387fc46e421dc43fec1c/common/app/views/support/DropdownMenus.scala#L43-L48

File: dotcom-rendering/cypress/e2e/parallel-3/article.metaAndOphan.e2e.cy.js [hunk 3]
```diff
@@ -52,16 +52,16 @@ describe('The web document renders with the correct meta and analytics elements
...
 		);
 		// Pillar ophan data-link-name exists with correct value
-		cy.get(`a[data-link-name="nav2 : primary : Opinion"]`).should(
+		cy.get(`a[data-link-name="nav3 : primary : Opinion"]`).should(
 			'have.length',
 			1,
 		);
 		// Only the top subnav is initially rendered so the count here is one
-		cy.get(`a[data-link-name="nav2 : subnav : money/pensions"]`).should(
+		cy.get(`a[data-link-name="nav3 : subnav : money/pensions"]`).should(
```
[the remark this answers is not in the document]
Comment (ioannakok) [reply dotcom-rendering/cypress/e2e/parallel-3/article.metaAndOphan.e2e.cy.js:60]: https://github.com/guardian/dotcom-rendering/pull/7893/files#diff-bfc476b74f778620278490b95b42ff291707ec06435c06418c4245784ebaf791R218-R221

[the remark this answers is not in the document]
Comment (mxdvl) [reply dotcom-rendering/cypress/e2e/parallel-3/article.metaAndOphan.e2e.cy.js:60]: Yes, for some reason my latest changes were not pushed. Should be fixed now

**Pull request merged**